cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -1290,11 +1327,15 @@ struct start_offset_output_iterator {
  private:
   __device__ reference dereference(int index)
   {
-    gpu::PageInfo const& p = pages[page_indices[index]];
-    if (p.src_col_schema != src_col_schema || p.flags & gpu::PAGEINFO_FLAGS_DICTIONARY) {
+    auto indices = get_indices{index, max_depth, num_pages};
```

Comment:
`const`?
I see that @PointKernel has pointed out a lot of places that need `const`. So please remember to add `const` wherever possible.